patch/kernel/archive/rockchip64-6.14/board-rocks0.patch

```diff
@@ -0,0 +1,176 @@
+* https://github.com/paolosabatino/armbian-build/blob/935149d2822cd2a2c63891ca6614563f5538b706/patch/kernel/archive/rockchip64-6.12/board-rocks0-0002-Revert-arm64-dts-rockchip-Fix-sdmmc-access-on-rk3308.patch
+---
+ .../boot/dts/rockchip/rk3308-rock-s0.dts      | 76 ++++++++++++++++++-
+ 1 file changed, 74 insertions(+), 2 deletions(-)
+
+diff --git a/arch/arm64/boot/dts/rockchip/rk3308-rock-s0.dts b/arch/arm64/boot/dts/rockchip/rk3308-rock-s0.dts
+index 8311af4c8689..2d102c482e3a 100644
+--- a/arch/arm64/boot/dts/rockchip/rk3308-rock-s0.dts
++++ b/arch/arm64/boot/dts/rockchip/rk3308-rock-s0.dts
+@@ -34,6 +34,22 @@ led-green {
+ 		};
+ 	};
+ 
++	acodec-sound {
++		compatible = "simple-audio-card";
++		simple-audio-card,format = "i2s";
++		simple-audio-card,name = "rockchip,rk3308-acodec";
++		simple-audio-card,mclk-fs = <256>;
++		simple-audio-card,codec-hp-det;
++		simple-audio-card,widgets =
++		"Headphone", "Headphones";
++		simple-audio-card,cpu {
++			sound-dai = <&i2s_8ch_2>;
++		};
++		simple-audio-card,codec {
++			sound-dai = <&codec>;
++		};
++	};
++
+ 	vdd_log: regulator-1v04-vdd-log {
+ 		compatible = "regulator-fixed";
+ 		regulator-name = "vdd_log";
+@@ -74,10 +90,24 @@ vcc_io: regulator-3v3-vcc-io {
+ 		vin-supply = <&vcc5v0_sys>;
+ 	};
+ 
++	vcc5v0_otg: vcc5v0-otg {
++		compatible = "regulator-fixed";
++		enable-active-high;
++		gpio = <&gpio0 RK_PC5 GPIO_ACTIVE_HIGH>;
++		pinctrl-names = "default";
++		pinctrl-0 = <&otg_vbus_drv>;
++		regulator-name = "vcc5v0_otg";
++		regulator-always-on;
++		vin-supply = <&vcc5v0_sys>;
++	};
++
+ 	/*
+ 	 * HW revision prior to v1.2 must pull GPIO4_D6 low to access sdmmc.
+ 	 * This is modeled as an always-on active low fixed regulator.
+ 	 */
++	/*
++	 * Armbian: this has been commented out due to consolidation of this patch:
++	 * https://github.com/paolosabatino/armbian-build/blob/935149d2822cd2a2c63891ca6614563f5538b706/patch/kernel/archive/rockchip64-6.12/board-rocks0-0002-Revert-arm64-dts-rockchip-Fix-sdmmc-access-on-rk3308.patch
+ 	vcc_sd: regulator-3v3-vcc-sd {
+ 		compatible = "regulator-fixed";
+ 		gpios = <&gpio4 RK_PD6 GPIO_ACTIVE_LOW>;
+@@ -90,6 +120,7 @@ vcc_sd: regulator-3v3-vcc-sd {
+ 		regulator-max-microvolt = <3300000>;
+ 		vin-supply = <&vcc_io>;
+ 	};
++	*/
+ 
+ 	vcc5v0_sys: regulator-5v0-vcc-sys {
+ 		compatible = "regulator-fixed";
+@@ -120,6 +151,11 @@ sdio_pwrseq: sdio-pwrseq {
+ 	};
+ };
+ 
++&codec {
++	status = "okay";
++	#sound-dai-cells = <0>;
++};
++
+ &cpu0 {
+ 	cpu-supply = <&vdd_core>;
+ };
+@@ -158,6 +194,22 @@ rtl8201f: ethernet-phy@1 {
+ 	};
+ };
+ 
++&i2c1 {
++	status = "okay";
++};
++
++&i2s_8ch_0 {
++	assigned-clocks = <&cru SCLK_I2S0_8CH_RX>;
++	assigned-clock-parents = <&cru SCLK_I2S0_8CH_TX_MUX>;
++	rockchip,clk-trcm = <1>;
++	#sound-dai-cells = <0>;
++};
++
++&i2s_8ch_2 {
++	status = "okay";
++	#sound-dai-cells = <0>;
```

Comment:
Normally the status property should be the last property, followed by any child nodes.
Also the `#sound-dai-cells` should probably have been added to the `codec` node in the soc dtsi.